index/index.go

```diff
@@ -341,8 +342,11 @@ func (w *Writer) AddSymbols(sym map[string]struct{}) error {
 
 	w.symbols = make(map[string]uint32, len(symbols))
 
-	for _, s := range symbols {
+	for index, s := range symbols {
 		w.symbols[s] = uint32(w.pos) + headerSize + uint32(w.buf2.len())
+		if w.Version == 2 {
```

Comment:
Unrelated to this change I suppose but do we really want the writer to be subject to versioning. I think we are fine with always have the writer write the most recent format version and just having the reader supporting multiple.

Reply:
I guess this is dependent on how we intend to support different index file versions in the future. Is the intention to only allow older versions for read purposed with the intention that they can load in an old tsdb database with a newer version of the code and it would then be written to an index file again with the new format?

If we don't intend to allow writing of index files with old versions then I'll remove the version checks from the writer.

Reply:
We don't want to write older versions.